semgrep-core/src/parsing/tree_sitter/Parse_php_tree_sitter.ml

```diff
@@ -291,6 +291,18 @@ let map_integer env tok =
   let value = int_of_string value in
   A.Int (Some value, tok)
 
+let map_boolean env tok =
+  let bool_of_string value =
+    let true_regexp = Str.regexp "['T''t']['R''r']['U''u']['E''e']" in
+    let false_regexp = Str.regexp "['F''f']['A''a']['L''l']['S''s']['E''e']" in
+    if Str.string_match true_regexp value 0 then true
+    else if Str.string_match false_regexp value 0 then false
+    else failwith ("Not a valid PHP boolean: " ^ value)
```

Comment:
Is there something less severe than a `failwith` that would be appropriate here?

Reply:
I don't think so :( I'll fix it later if it turns out there is